test-utils/src/main/java/com/ibm/ws/lars/testutils/clients/LooseFileWriteableClient.java

```diff
@@ -0,0 +1,95 @@
+/*******************************************************************************
+* Copyright (c) 2017 IBM Corp.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+package com.ibm.ws.lars.testutils.clients;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Collection;
+
+import com.ibm.ws.repository.transport.client.DirectoryUtils;
+import com.ibm.ws.repository.transport.client.LooseFileClient;
+import com.ibm.ws.repository.transport.exceptions.BadVersionException;
+import com.ibm.ws.repository.transport.exceptions.RequestFailureException;
+import com.ibm.ws.repository.transport.model.Asset;
+import com.ibm.ws.repository.transport.model.Attachment;
+import com.ibm.ws.repository.transport.model.AttachmentSummary;
+
+/**
+ *
+ */
+public class LooseFileWriteableClient extends AbstractFileWriteableClient {
```

Comment:
How exactly does this class work?

When an asset is created, presumably we need to create a file for it? And then we need somewhere to put that file? I would expect to see a base directory or something on this class to control where it's going to create files, and also some code that is making up a filename to use for the asset.

I would appreciate some javadoc explaining how this class is meant to work. The LooseFileClient is inherently read only, you're giving it a set of classes and saying "here's your metadata". I'd assume that the writable client would somehow create files and then pass those in, which then means the regular LooseFileClient needs some way of having its collection of files updated, which isn't terribly nice.

Reply:
Base directory added and override the base classes code that decides upon a filename. The writeable and readable clients share the same backing collection (which isn't nice I agree but seemed liked the easiest way for creating the tests).